aws/acm.go

```diff
@@ -0,0 +1,213 @@
+package aws
+
+import (
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/acm"
+	"github.com/aws/aws-sdk-go/service/acm/acmiface"
+)
+
+const glob = "*"
+
+type certificateCache struct {
+	sync.Mutex
+	acmClient      acmiface.ACMAPI
+	acmCertSummary []*acm.CertificateSummary
+	acmCertDetail  []*acm.CertificateDetail
+}
+
+func NewCertCache(acmClient acmiface.ACMAPI) *certificateCache {
+	return &certificateCache{
+		acmClient:      acmClient,
+		acmCertSummary: make([]*acm.CertificateSummary, 0),
+		acmCertDetail:  make([]*acm.CertificateDetail, 0),
+	}
+}
+
+func (cc *certificateCache) InitCertCache(certUpdateInterval time.Duration) {
+	go func() {
+		for {
+			log.Println("update cert cache")
+			cc.updateCertCache()
+			time.Sleep(certUpdateInterval)
+		}
+	}()
+
+}
+
+// updateCertCache will only update the current certificateCache if
+// all calls to AWS API were successfull and not rateLimited for
+// example
+func (cc *certificateCache) updateCertCache() {
+	certList, err := cc.ListCerts()
+	if err != nil {
+		log.Printf("Could not update certificate cache, caused by: %v", err)
+		return
+	}
+
+	certDetails := make([]*acm.CertificateDetail, len(certList), len(certList))
+	for idx, o := range certList {
+		certInput := &acm.DescribeCertificateInput{CertificateArn: o.CertificateArn}
+		certDetail, err := cc.acmClient.DescribeCertificate(certInput)
+		if err != nil {
+			log.Printf("Could not get certificate details from AWS for ARN: %s, caused by: %v", o.CertificateArn, err)
+			return
```

Comment:
Hmm, a `break` could be more appropriate. I can see how this could help with consistency but, is it really an issue? How different would it be from an outdated cache?